tests/library/conftest.py

```diff
@@ -0,0 +1,193 @@
+import logging
+import os
+from typing import Optional
+
+from pytest import MonkeyPatch
+from rdflib import Graph
+from rdflib.namespace import NamespaceManager
+from sqlalchemy import create_engine
+from sqlalchemy.orm import scoped_session, sessionmaker
+
+import buildingmotif
+import buildingmotif.building_motif.building_motif
+import buildingmotif.building_motif.singleton
+from buildingmotif.building_motif.building_motif import BuildingMotifEngine
+from buildingmotif.database.graph_connection import GraphConnection
+from buildingmotif.database.table_connection import TableConnection
+from buildingmotif.database.tables import Base as BuildingMOTIFBase
+from buildingmotif.database.utils import (
+    _custom_json_deserializer,
+    _custom_json_serializer,
+)
+from buildingmotif.namespaces import bind_prefixes
+
+# The fact that BuildingMOTIF is a singleton class is a problem for testing templates.
+# We want to have tests which are parameterized by the library and template name; this makes
+# it possible to use pytest filters (with the "-k" flag) to run tests for specific libraries or templates.
+# We want each (library, template) pair to operate in a "clean" environment, so that the tests are isolated.
+# However, the singleton pattern means that the BuildingMOTIF instance is shared across all tests.
+# We can work around this by patching BuildingMOTIF to ignore the singleton pattern.
+
+# "instances" is a dictionary that maps the name of the module to the BuildingMOTIF instance.
+instances = {}
+
+
+# non-singleton BuildingMOTIF
+class BuildingMOTIF:
+    """Manages BuildingMOTIF data classes."""
```

Comment:
I think this can be avoided by the following:
You _can_ create multiple instances of `BuildingMOTIF` if you call `BuildingMOTIF.clean()` between each constructor call. This only deletes the current instance from the class object, you can still keep it and do stuff with it.

Reply:
!!! Awesome

Reply:
One additional thing I just thought of, you may be able to avoid the monkeypatch stuff for `get_building_motif` by just setting the `instance` attribute on the `BuildingMOTIF` class to the one you want to use.